src/internal/internal_gemmA.cc

```diff
@@ -81,31 +82,36 @@ void gemmA(internal::TargetType<Target::HostTask>,
     assert( A.nt() == B.mt() );
     assert( A.mt() == C.mt() );
 
+    const scalar_t zero = 0.0;
+    const scalar_t one  = 1.0;
+
     int err   = 0;
     // This assumes that if a tile has to be acquired, then all tiles
     // have to be acquired
     // TODO make it a matrix of the C tiles involved c.TileAcquire(i, k)
     int c_tile_acquired = 0;
     #pragma omp taskgroup
     for (int64_t i = 0; i < A.mt(); ++i) {
+        int nlocal_A_tiles = 0;
         for (int64_t j = 0; j < A.nt(); ++j) {
             if (A.tileIsLocal( i, j )) {
+                nlocal_A_tiles++;
                 #pragma omp task slate_omp_default_none \
                     shared( A, B, C, err, c_tile_acquired ) \
                     firstprivate(i, j, layout) priority(priority)
                 {
                     try {
-                        A.tileGetForReading( i, j, LayoutConvert( layout ) );
+                        A.tileGetForReading( i, j, HostNum, LayoutConvert( layout ) );
                         for (int64_t k = 0; k < B.nt(); ++k) {
                             B.tileGetForReading(
-                                j, k, LayoutConvert( layout ) );
+                                j, k, HostNum, LayoutConvert( layout ) );
 
                             if (C.tileIsLocal( i, k )) {
                                 C.tileGetForWriting(
-                                    i, k, LayoutConvert( layout ) );
+                                    i, k, HostNum, LayoutConvert( layout ) );
                             }
                             else {
-                                if (! C.tileExists( i, k )) {
+                                if (! C.tileExists( i, k, HostNum )) {
                                     c_tile_acquired = 1;
                                     #pragma omp critical
                                     {
```

Comment:
Yes, I do not understand it either. I can try to remove it.

Reply:
So far, no issue when removing it. But I suspect it may be related to the `layout`.
When `tileAcquire` is called, only the inner routine `tileAcquire` is protected by the lock. Then if the tile layout is different than `layout`, I think it is unsafe to do the `tileMakeTransposable` (see [BaseMatrix](https://github.com/icl-utk-edu/slate/blob/master/include/slate/BaseMatrix.hh#L2709)).
So I think we should discuss either it is fine to remove it or add the `omp critical` every time we call tileAcquire.

Reply:
Just leave for now. We can revisit locking later.